operator/pkg/manifest/client.go

```diff
@@ -57,78 +55,85 @@ func NewHelmClient(kubeClient *kube.Client, restGetter *manifestRest.ManifestRES
 	// Use deferred discovery client here as GVs applicable to the client are inconsistent at this moment
 	discoveryMapper := restmapper.NewDeferredDiscoveryRESTMapper(discoveryClient)
 
-	return &HelmClient{
-		kubeClient: kubeClient,
-		settings:   settings,
-		restGetter: restGetter,
-		clientSet:  clientSet,
-		restConfig: restConfig,
-		mapper:     discoveryMapper,
-	}, nil
-}
+	helmClient := &HelmClient{
+		logger:      logger,
+		repoHandler: NewRepoHandler(logger, settings),
+		settings:    settings,
+		restGetter:  restGetter,
+		restConfig:  restConfig,
+		mapper:      discoveryMapper,
+	}
 
-func (h *HelmClient) getGenericConfig(namespace string) (*action.Configuration, error) {
-	actionConfig := new(action.Configuration)
-	if err := actionConfig.Init(h.restGetter, namespace, "secrets",
-		func(format string, v ...interface{}) {
-			callDepth := 2
-			format = fmt.Sprintf("[debug] %s\n", format)
-			err := log.Output(callDepth, fmt.Sprintf(format, v...))
-			if err != nil {
-				log.Println(err.Error())
-			}
-		}); err != nil {
+	helmClient.actionClient, helmClient.kubeClient, err =
+		helmClient.newInstallActionClient(v1.NamespaceDefault, restGetter)
```

Comment:
considering that you always call this with default namespace this could be cached together with the helmclient for every run. You probably dont need to reinitialize it

Reply:
this is cached as this is a part of HelmClient